### PR TITLE
SDIT-1571 Create alert endpoint

### PR DESCRIPTION
**Pull request by andymarke**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/alerts/AlertsResource.kt [hunk 1]
```diff
@@ -137,3 +208,36 @@ data class NomisAudit(
   @Schema(description = "Additional information that is audited")
   val auditAdditionalInfo: String? = null,
 )
+
+@Schema(description = "A request to create an alert in NOMIS")
+@JsonInclude(JsonInclude.Include.NON_NULL)
+data class CreateAlertRequest(
+  @Schema(description = "The alert code")
+  @NotNull
```
Comment (andymarke) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/alerts/AlertsResource.kt:216]: TODO - forgot how do get decent validation error messages for non-null attributes, this is nonsense but I will come back to it

File: src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/jpa/OffenderAlert.kt [hunk 2]
```diff
@@ -98,6 +98,9 @@ class OffenderAlert(
   @OneToMany(mappedBy = "alert", fetch = LAZY, cascade = [CascadeType.ALL])
   var workFlows: MutableList<AlertWorkFlow> = mutableListOf(),
 
+  @Column(name = "CREATE_USER_ID")
+  var createUsername: String,
+
```
Comment (andymarke) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/jpa/OffenderAlert.kt:103]: first time we are propagating the create username  for sync, not sure if we need to  ...

File: src/test/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/alerts/AlertsResourceIntTest.kt [hunk 3]
```diff
@@ -334,6 +336,411 @@ class AlertsResourceIntTest : IntegrationTestBase() {
           .jsonPath("audit.auditClientWorkstationName").isEqualTo("MMD1234J")
           .jsonPath("audit.auditAdditionalInfo").isEqualTo("POST /api/bookings/2904199/alert")
       }
+
+      @Test
+      fun `can use both GET Urls`() {
+        webTestClient.get().uri("/prisoner/booking-id/$bookingId/alerts/$alertSequenceWithAuditMinimal")
```
Comment (andymarke) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/alerts/AlertsResourceIntTest.kt:342]: moving to a `/prisoners/` url for consistency (keeping old one for a bit)

File: src/test/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/helper/builders/OffenderAlert.kt [hunk 4]
```diff
@@ -161,6 +161,7 @@ class OffenderAlertBuilder(
     alertStatus = status,
     commentText = commentText,
     verifiedFlag = verifiedFlag,
+    createUsername = "SA",
   ).apply {
```
Comment (andymarke) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/helper/builders/OffenderAlert.kt:165]: was default when running tests

**Review by lbennett-moj — APPROVED** [review]

**Pull request merged**